spring-boot-project/spring-boot-tools/spring-boot-configuration-processor/src/main/java/org/springframework/boot/configurationprocessor/ConfigurationMetadataAnnotationProcessor.java

```diff
@@ -353,6 +356,22 @@ private ConfigurationMetadata mergeAdditionalMetadata(ConfigurationMetadata meta
 		return metadata;
 	}
 
+	private void validateDefaultValuesResolved(ConfigurationMetadata metadata) {
+        metadata.getItems().forEach((itemMetadata -> {
+			if (itemMetadata.getDefaultValue() instanceof ValueWrapper) {
```

Comment:
I have not propagated the change from `Object -> ValueWrapper` up from FVP to `ItemMetadata` yet but will do that shortly. Unless you can think of a reason to keep it `Object`.

Reply:
One reason to keep it Object is to prevent the ripple around all areas that are accessing ItemMetadata#defaultValue (JsonConverter, MetadataStore, etc..).

Reply:
Yeah. I didn't even think of changing `ItemMetadata` to be honest. I was hoping we could trigger the warning/error locally to field resolution by passing the necessary bits to it.

Reply:
Unfortunately we have to wait until the additional metadata is merged to see if the user added a manual hint.